lib/deprecated/misc.nix

```diff
@@ -329,6 +329,7 @@ let
   fakeHash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
   fakeSha256 = "0000000000000000000000000000000000000000000000000000000000000000";
   fakeSha512 = "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
+  randomHash = builtins.hashString "sha256" (builtins.readFile /proc/sys/kernel/random/uuid);
```

Comment:
This cannot take effect in flakes I suppose?

Reply:
I guess not, and there's no `--argstr` to pass it from shell either https://github.com/NixOS/nix/issues/5663

Some alternative approaches could be
- external script driving multiple builds and substituting fakeHash per invocation & package
- tweaking Nix to allow FOD without outputHash, it will fail but won't wait on other FODs
- tweaking Nix to treat lib.fakeHash FOD specially, so not locking
- replacing FOD outputHash with completely different flow for things that aren't just fetches
  - replace pinning outputHash with trush of substitution / attestation that a given unsafe derivation should produce given output
  - maybe distinguish between output value trust and trust to allow building it without full sandbox
  - that should make maintaining packages simpler, Hydra would sign outputs and signatures aren't persisted in nixpkgs, consumers may decide to trust Hydra or take the risk of building, but package maintainer doesn't need to do anything
- having pseudo-random hash where sources of entropy could be
  - user-supplied salt
  - system argument (not sure how it works with flakes)
  - package name
  - current Git commit or tree hash (in case of flakes)
- having multiple fakeHash values
 -  weaker version of user salt
 - still locks for different systems unless values are per-system
 - still locks if several builds reuse the value

Reply:
> tweaking Nix to treat lib.fakeHash FOD specially, so not locking

Probably the only correct way.